rcl/src/rcl/service.c

```diff
@@ -280,6 +280,21 @@ rcl_send_response(
   return RCL_RET_OK;
 }
 
+bool
+rcl_service_is_valid(const rcl_service_t * service)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(service, false, rcl_get_default_allocator());
+  RCL_CHECK_FOR_NULL_WITH_MSG(service->impl,
+    "rcl service implementation is invalid",
+    return false,
+    rcl_get_default_allocator());
+  RCL_CHECK_FOR_NULL_WITH_MSG(service->impl->rmw_handle,
+    "rmw handle is invalid",
+    return false,
+    rcl_get_default_allocator());
+  return true;
+}
```

Comment:
Why does this one not check the options of the service like it does for client and pub/sub?